tests/integration/scheduler.bats

```diff
@@ -12,17 +12,51 @@ function teardown() {
 }
 
 @test "scheduler is applied" {
-	update_config ' .process.scheduler = {"policy": "SCHED_DEADLINE", "nice": 19, "priority": 0, "runtime": 42000, "deadline": 1000000, "period": 1000000, }'
+	update_config ' .process.scheduler = {
+		"policy": "SCHED_BATCH",
+		"priority": 0,
+		"nice": 19
+	}'
 
 	runc run -d --console-socket "$CONSOLE_SOCKET" test_scheduler
 	[ "$status" -eq 0 ]
 
+	# Check init settings.
 	runc exec test_scheduler chrt -p 1
 	[ "$status" -eq 0 ]
+	[[ "${lines[0]}" == *"scheduling policy: SCHED_BATCH" ]]
+	[[ "${lines[1]}" == *"priority: 0" ]]
+
+	# Check exec settings derived from config.json.
+	runc exec test_scheduler sh -c 'chrt -p $$'
+	[ "$status" -eq 0 ]
+	[[ "${lines[0]}" == *"scheduling policy: SCHED_BATCH" ]]
+	[[ "${lines[1]}" == *"priority: 0" ]]
+
+	# Another exec, with different scheduler settings.
+	proc='
+{
+	"terminal": false,
+	"args": [ "/bin/sleep", "600" ],
+	"cwd": "/",
+	"scheduler": {
+		"policy": "SCHED_DEADLINE",
+		"flags": [ "SCHED_FLAG_RESET_ON_FORK" ],
+		"nice": 19,
+		"priority": 0,
+		"runtime": 42000,
+		"deadline": 100000,
+		"period": 1000000
+	}
+}'
+	__runc exec -d --pid-file pid.txt --process <(echo "$proc") test_scheduler
+	[ "$status" -eq 0 ]
 
-	[[ "${lines[0]}" == *"scheduling policy: SCHED_DEADLINE" ]]
+	# shellcheck disable=SC2016
+	run chrt -p "$(cat pid.txt)"
```

Comment:
I can't see why this run line will trigger that shelcheck: https://www.shellcheck.net/wiki/SC2016

Do you understand why?

Reply:
I think this is a leftover from when I was doing things in a different way.

Removed now, thanks for catching this.

Reply:
Thanks!